### DISCUSSION TITLE
Telekom Slovenia IPTV

### DISCUSSION THREAD

**Discussion by MermenJazz** [fabianishere/udm-iptv#185] [Q&A]
Hello,

I just got my UDM SE and I'm trying to set up IPTV. My ISP is Telekom Slovenia, PPPoE, VLAN 3999 for IPTV.
I apologize in advance, I've just started learning about networking, please keep that in mind.

I previously had an ASUS router (AC68U) and it worked with just creating a VLAN and tagging the WAN (to the ISPs modem) and LAN port (where my STB is connected). No other config was needed to make it work.

With the old setup IPTV worked even if PPPoE was down. There was no DHCP for VLAN 3999, devices on this VLAN would get a 169.254.x.x IP and IPTV worked (EDIT: valid for PCs, STBs DO get an IP, see below).

Now I took the SFP module from my ISPs modem and put it in port 10 of the UDM SE. 

This is the diag command output:

```
=== Configuration ===
WAN Interface: eth9
WAN VLAN: 3999 (dev eth9)
WAN DHCP: false (options "")
WAN Ranges: 232.2.0.0/16 232.4.0.0/16
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
=== Service Logs ===
Sep 24 10:41:53 UDMSE udm-iptv[18603]: Device eth9 already exists.. Deleting device
Sep 24 10:41:53 UDMSE udm-iptv[18603]: RTNETLINK answers: Operation not supported
Sep 24 10:41:53 UDMSE udm-iptv[18603]: RTNETLINK answers: File exists
Sep 24 10:41:53 UDMSE udm-iptv[18603]: NATing IPTV network ranges (if necessary)
Sep 24 10:41:53 UDMSE udm-iptv[18603]: Setting up igmpproxy
Sep 24 10:41:53 UDMSE udm-iptv[18603]: Starting igmpproxy
Sep 24 10:41:53 UDMSE igmpproxy[18603]: There must be at least 1 Vif as upstream.
Sep 24 10:41:53 UDMSE systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Sep 24 10:41:53 UDMSE systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Sep 24 10:41:53 UDMSE systemd[1]: udm-iptv.service: Consumed 92ms CPU time.

```

I tried with a IPTV network with DHCP, a VLAN-only network, changing the WAN interface to ppp0, sadly nothing worked.
Would anyone have an idea on how to configure it with this setup?

Thank you very much,
MermenJazz

Edit: I just now discovered that the STBs actually DO get an IP (10.64.184.xxx). Digging around the webUI of my modem, under Static Route, I found this:
DstIP/PrefixLength 10.253.9.6/32
Gateway 10.64.184.1
Interface eth5.2 (should be the SFP port on the modem)

**Comment by fabianishere** [comment]
You might need custom DHCP options to obtain a lease from your ISP.

**Reply by MermenJazz** [reply]
I got some channels working last night with setting a random static IP, but only from 1 source IP.
All the source IPs are in the 95.176.x.x range and destination IPs in 232.2.x.x and 232.4.x.x (the last 2 being the IPs I have in my playlist). Got the IPs with Wireshark on my PC.

I had to manually create a link (eth9.3999) for the WAN port with VLAN 3999.

```
ip link add link eth9 name eth9.3999 type vlan id 3999
ip addr add 10.10.10.10/32 dev eth9.3999
```

Could I sniff what the STB is doing to get an IP with Wireshark or something like that?
My ISP isn't willing to share any details with me.

**Reply by fabianishere** [reply]
Yes, you can use tcpdump to capture the packets on your UDM Pro. See the following comment: https://github.com/fabianishere/udm-iptv/issues/123#issuecomment-1134582612

You can then open the resulting pcap file in Wireshark

**Reply by MermenJazz** [reply]
While waiting for your response I connected my PC to the STB and listened to the virtual ethernet adapter with tagged 3999 VLAN.

I found the DHCP request with the following options:
53, 57, 60, 12, 55

I'll try to add them to IPTV_WAN_DHCP_OPTIONS and report back.

To output on VLAN 3999 on the LAN side, should I set IPTV_LAN_INTERFACES="br3999"?

Thanks for taking your time to help, much appreciated!

Edit: SUCCESS!
After adding some DHCP options I got a lease in the same range as the STB!

`-O staticroutes -V Device name -x hostname:Devicename-xxxxxx`

**Reply by fabianishere** [reply]
Correct, if you create the network in the UniFi interface beforehand.

**Reply by fabianishere** [reply]
For the DHCP options, option 60 (vendor class identifier) is probably important to have your router identified as IPTV receiver by your ISP.

**Reply by MermenJazz** [reply]
Correct, needed the vendor class identifier and hostname (wouldn't work without the last one).
I now get the route I saw in my modem's webUI: 10.253.0.0/16 via 10.96.184.1 (presumably with -O staticroutes?)

I'm still having trouble playing all the channels in the playlist. I only get channels that come from 95.176.249.18, but there's more sources in the 95.176.x.x subnet. Any idea what could cause this?

**Reply by fabianishere** [reply]
Usually this is an issue with your WAN ranges. Probably you are missing some ranges.

**Reply by MermenJazz** [reply]
I have it at 0.0.0.0/0 at the moment.

I'll try and do some more testing. Please let me know if you have any ideas.
Thanks for your time, really appreciate it!

**Reply by MermenJazz** [reply]
I'm having trouble with br3999 and the VLAN settings in UniFi. Can't get it to output on the tagged VLAN interfaces. Set a VLAN with DHCP, tried with switch profiles with tagged VLAN 3999 only and no native network, native 3999, tagged 3999 only.

What am I missing?! Spent hours searching on the internet, but I'm stuck. Any tips?

It does work on br0 and an untagged VLAN! And I do restart the service every time I change settings.

**Reply by fabianishere** [reply]
Are you sure “LAN interfaces” is set correctly?

**Reply by MermenJazz** [reply]
Other than setting br3999 in the config is there anything else? Or maybe some UniFi settings to make tagged VLANs work that are not on by default?

**Reply by fabianishere** [reply]
Does `udm-iptv-diag` show `br3999` for "LAN interfaces"?

**Reply by MermenJazz** [reply]
Yes, it does!

```
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 3999 (dev eth8.3999)
WAN DHCP: true (options "-O staticroutes -x hostname:TViNBox-282875")
WAN Ranges: 0.0.0.0/0
LAN Interfaces: br3999
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
36: eth8.3999@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.96.185.52/21 brd 10.96.191.255 scope global eth8.3999
       valid_lft forever preferred_lft forever
10.96.184.0/21 proto kernel scope link src 10.96.185.52
10.253.0.0/16 via 10.96.184.1 metric 236
=== Service Logs ===
Sep 28 18:06:55 UDMSE systemd[1]: udm-iptv.service: Consumed 1.876s CPU time.
Sep 28 18:06:55 UDMSE systemd[1]: Started IPTV support for the UniFi Dream Machine.
Sep 28 18:06:55 UDMSE udm-iptv[26005]: Obtaining IP address for VLAN interface
Sep 28 18:06:55 UDMSE udm-iptv[26005]: udhcpc (v1.22.1) started
Sep 28 18:06:55 UDMSE udm-iptv[26005]: Sending discover...
Sep 28 18:06:56 UDMSE udm-iptv[26005]: Sending select for 10.96.185.52...
Sep 28 18:06:56 UDMSE udm-iptv[26005]: Lease of 10.96.185.52 obtained, lease time 120745
Sep 28 18:06:56 UDMSE udm-iptv[26005]: NATing IPTV network ranges (if necessary)
Sep 28 18:06:56 UDMSE udm-iptv[26005]: Setting up igmpproxy
Sep 28 18:06:56 UDMSE udm-iptv[26005]: Starting igmpproxy
```

**Reply by MermenJazz** [reply]
I just restarted it and now it works... Really weird!

Now the only thing is to figure out why I only get channels from one of the source IPs and not the other ones...
WAN Ranges at 0.0.0.0/0.  What else would impact this? Any clues?

Also just noticed a working channels stops working after aprox. 30s...

**Reply by MermenJazz** [reply]
Does this config use IGMPv2 or V3? Can be v3 be enabled if it's not already?
Looks like my STB is using v3, could that be the problem?

**Reply by fabianishere** [reply]
I don’t believe ‘igmpproxy’ supports igmpv3 at the moment.

**Reply by MermenJazz** [reply]
Thanks for your reply.
I got it working momentarily after adding in more DHCP options. Streams still freeze after 30s tho.

If I get it working 100% I'll post the config here.

**Comment by MermenJazz** [comment]
In the end I managed to make it work!

1. Created a VLAN-only network with VLAN ID 3999.
2. SSH:
```
ip link add link eth8 name eth8.3999 type vlan id 3999

ip link set dev eth8.3999 master br3999

ip link set dev eth8.3999 up
```

Now I think I need to isolate VLAN 3999 from the rest of my network with some firewall rules and that should be all. (?)

Any suggestions? I'm a noob who just googles a lot, so criticism is appreciated!

Edit: I'm currently connected on eth8 or 2.5Gb WAN to the modem, will move it to eth9 and plug the fiber directly in the UDM tomorrow!

**Reply by MermenJazz** [reply]
Changed a setting in the web ui and the ssh commands didn't survive. What's the best and simplest way to have them survive provisioning and reboot?
